packages/actions-spec/README.md

```diff
@@ -225,20 +235,160 @@ export interface LinkedAction {
   /** button text rendered to the user */
   label: string;
   /** Parameter to accept user input within an action */
-  parameters?: [ActionParameter];
+  parameters?: Array<ActionParameter | ActionParameterSelectable>;
```

Comment:
Given the d.ts file now uses a new type, should this be updated as well?

Reply:
I added a note to the top of this readme (and evnetually to the published docs) that the types/interfaces shown in here may be simplified version of the final.

do you think it is worth adding the `TypedActionParameter` definition into the readme?

Reply:
and since a dev could also still access these specific types based on the parent definition (like `ActionParameter<"email">["min"]`) they are still accessible. This way they will also get the correct type based on the parameter `type` too

Reply:
Ah, makes sense. Yeah, that's fine, just concerned that it's likely that a dev would copy-paste or look for certain types based on the doc. So the closer the better.

We can also do something like:

```ts
/**
 @see {TypedParameter}
*/
parameters?: Array<ActionParameter | ActionParameterSelectable>;
```

or the other way around:

```ts
/**
  @see {ActionParameter}
  @see {ActionParameterSelectable}
*/
parameters?: Array<TypedParameter>;
```

The closer to the actual types the better, but I agree - simplified is good

Reply:
I like the `@see {TypedParameter}` idea. Let's go with that!

I could add a more explicit message too, something like:
```
/**
   @see `TypedParameter` for the non-simplified type
*/
parameters?: Array<ActionParameter | ActionParameterSelectable>;
```

That way we get the best of both worlds I think. Simplified in the readme, but being explicit for people when they read it.

I would be super surprised if people tried to copy pasta the type/interfaces out of the doc since there is not really a reason to do so. But I guess you never know

Reply:
I like this option:

```
/**
  @see {ActionParameter}
  @see {ActionParameterSelectable}
*/
parameters?: Array<TypedParameter>;
```

That way we get the best of both worlds I think. Simplified in the readme, but being explicit for people when they read it.

I would be super surprised if people tried to copy pasta the type/interfaces out of the doc since there is not really a reason to do so. But I guess you never know